include/deal.II/numerics/vector_tools.h

```diff
@@ -857,6 +858,59 @@ namespace VectorTools
                 const bool                          project_to_boundary_first = false);
 
   /**
+   * The same as above for projection of scalar-valued quadrature data in case
+   * of parallel Triangulations.
+   * The underlying MatrixFree implementation requires to specify the degree of
+   * the used FiniteElement as additional template parameters.
+   * The user provided function should return a value at the quadrature point
+   * based on the cell iterator and quadrature number and of course should be
+   * consistent with the provided @p quadrature object, which will be used
+   * to assemble the right-hand-side.
+   *
+   * This function can be used with lambdas:
+   * @code
+   * VectorTools::project_parallel<dim,LA::MPI::Vector,dim,fe_degree>
+   * (mapping,
+   *  dof_handler,
+   *  constraints,
+   *  quadrature_formula,
+   *  [=] (const typename DoFHandler<dim>::active_cell_iterator & cell, const unsigned int q) -> double { return qp_manager.get_data(cell)[q]->density; },
```

Comment:
your example is not exactly illustrative in that you don't use quadrature point argument, i will specify in the documentation what qp_manager is.